ferry.go

```diff
@@ -50,6 +50,23 @@ func MaskedDSN(c *mysql.Config) string {
 	return c.FormatDSN()
 }
 
+type SafeState struct {
+	state string
+	mutex sync.Mutex
+}
+
+func (s *SafeState) Value() string {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	return s.state
+}
+
+func (s *SafeState) Update(state string) {
+	s.mutex.Lock()
+	s.state = state
+	s.mutex.Unlock()
+}
```

Comment:
👌 

You can use the `atomic` package (`atomic.Value`) if you wanted to simplify this

Reply:
I don't see the `sync/atomic` package dealing with strings?

Reply:
`atomic.Value` deals with any type, just need to `Load` and `Store` a consistent type across calls

Reply:
I see, let me take a look.